sdk/python/lib/pulumi/provider/server.py

```diff
@@ -71,6 +112,19 @@ async def Construct(  # pylint: disable=invalid-overridden-method
         await self.lock.acquire()
         try:
             return await self._construct(request, context)
+        except InputPropertiesError as e:
+            status = self.create_grpc_invalid_properties_status(e.message, e.errors)
+            await context.abort_with_status(status)
+            # We already aborted at this point
+            raise
+        except InputPropertyError as e:
+            status = self.create_grpc_invalid_properties_status(
+                "", [{"property_path": e.property_path, "reason": e.reason}]
+            )
+            await context.abort_with_status(status)
+            # We already aborted at this point
+            raise
```

Comment:
It occurred to me when reviewing this PR: we should probably do the same treatment in `Call` as well. But that can be done as a follow-up for all of the SDKs.

Reply:
Yeah, let's do that separately.